crates/oxc_parser/src/jsx/mod.rs

```diff
@@ -246,11 +247,12 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_jsx_assignment_expression(&mut self) -> Result<Expression<'a>> {
-        // TODO: check for SequenceExpression and warn
-        // "JSX expressions may not use the comma operator. Did you meant to write an array?"
         let ctx = self.ctx;
         self.ctx = Context::default();
         let expr = self.parse_expression();
+        if let Ok(Expression::SequenceExpression(seq)) = &expr {
+            return Err(diagnostics::JSXExpressionsMayNotUseTheCommaOperator(seq.span).into());
+        }
```

Comment:
Not sure if this is what you had in mind with "warn"? Should we error here?

Reply:
Yes it should be an error.